tests/functional/base.py

```diff
@@ -0,0 +1,316 @@
+
+    def lease_create(self, client, node, lessee, parse=True, **kwargs):
+        flags_default = { 'resource_type': 'dummy_node' }
+        for flag in 'start_time', 'end_time', 'name', 'properties':
+            flags_default[flag] = None
+
+        flags = self._kwargs_to_flags(self._merge_kwargs(flags_default, kwargs))
+        output = self.clients[client].esi('lease create', flags, '%s %s' %
+                (node['uuid'], self.client_info[lessee]['project_name']))
+
+        return self.parse_details(output) if parse else output
+
+    def lease_list(self, client, parse=True, **kwargs):
+        flags_default = {}
+        for flag in ('long', 'all', 'status', 'offer_uuid', 'time_range',
+                'project', 'owner', 'resource_type', 'resource_uuid'):
+            flags_default[flag] = None
+
+        flags = self._kwargs_to_flags(self._merge_kwargs(flags_default, kwargs))
+        output = self.clients[client].esi('lease list', flags, '')
+
+        return self.parser.listing(output) if parse else output
+
+    def lease_delete(self, client, lease_uuid):
+        return self.clients[client].esi('lease delete', '', lease_uuid)
+
+    def lease_show(self, client, lease_uuid, parse=True):
+        output = self.clients[client].esi('lease show', '', lease_uuid)
+        return self.parse_details(output) if parse else output
+
+    def owner_change_create(self, client, owner_prev, owner_new, node, 
+            parse=True, **kwargs):
+        flags_default = { 'resource_type': 'dummy_node' }
+        for flag in 'start_time', 'end_time':
+            flags_default[flag] = None
+
+        flags = self._kwargs_to_flags(self._merge_kwargs(flags_default, kwargs))
+        output = self.clients[client].esi('owner change create', flags,
+                '%s %s %s' % (self.client_info[owner_prev]['project_id'],
+                    self.client_info[owner_new]['project_id'], node['uuid']))
+
+        return self.parse_details(output) if parse else output
+
+    def owner_change_list(self, client, parse=True, **kwargs):
+        flags_default = {}
+        for flag in ('long', 'status', 'time_range', 'from_owner_id',
+                'to_owner_id', 'resource_uuid', 'resource_type'):
+            flags_default[flag] = None
+
+        flags = self._kwargs_to_flags(self._merge_kwargs(flags_default, kwargs))
+        output = self.clients[client].esi('owner change list', flags, '')
+
+        return self.parser.listing(output) if parse else output
+
+    def owner_change_delete(self, client, owner_change_uuid):
+        return self.clients[client].esi('owner change delete', '',
+                owner_change_uuid)
+
+    def owner_change_show(self, client, owner_change_uuid, parse=True):
+        output = self.clients[client].esi('owner change show',
+                '', owner_change_uuid)
+        return self.parse_details(output) if parse else output
+
+class EsiCliClient(base.CLIClient):
```

Comment:
'ESICLIClient' - I know it looks weird :)

Reply:
The only reason I wasn't capitalizing ESI before was because ESICLIClient looks goofy lol (will be fixed)